src/main/java/org/rust/lang/core/lexer/LexerBaseKt.java

```diff
@@ -0,0 +1,61 @@
+package org.rust.lang.core.lexer;
+
+import com.intellij.lexer.LexerBase;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Small utility class to ease implementing {@link LexerBase} in Kotlin.
+ */
+public abstract class LexerBaseKt extends LexerBase {
+    private int state;
+    private int tokenStart;
+    private int tokenEnd;
+    private CharSequence bufferSequence;
+    private int bufferEnd;
+
+    @Override
+    public int getState() {
+        return state;
+    }
+
+    protected void setState(int state) {
+        this.state = state;
+    }
+
+    @Override
+    public int getTokenStart() {
+        return tokenStart;
+    }
+
+    protected void setTokenStart(int tokenStart) {
+        this.tokenStart = tokenStart;
+    }
+
+    @Override
+    public int getTokenEnd() {
+        return tokenEnd;
+    }
+
+    protected void setTokenEnd(int tokenEnd) {
+        this.tokenEnd = tokenEnd;
+    }
+
+    @NotNull
+    @Override
+    public CharSequence getBufferSequence() {
+        return bufferSequence;
+    }
+
+    protected void setBufferSequence(CharSequence bufferSequence) {
+        this.bufferSequence = bufferSequence;
+    }
+
+    @Override
+    public int getBufferEnd() {
+        return bufferEnd;
+    }
+
+    protected void setBufferEnd(int bufferEnd) {
+        this.bufferEnd = bufferEnd;
+    }
+}
```

Comment:
Kotlin sees overrided methods from `Lexer` as read-only properties, while we need setters for these properties in our lexer. I didn't know how to do this in Kotlin (probably it's impossible), so I falled back to Java here. Of course we can omit this class if we declare custom properties for all this stuff in `RustStringLiteralLexer`, but it's fishy imho